brigade/core/__init__.py

```diff
@@ -36,13 +36,27 @@ def _unpickle_method(func_name, obj, cls):
 logger = logging.getLogger("brigade")
 
 
+class Data(object):
+    """
+    This class is just a placeholder to share data amongsts different
+    versions of Brigade  after running ``filter`` multiple times.
+
+    Attributes:
+        failed_hosts (list): Hosts that have failed to run a task properly
+    """
+
+    def __init__(self):
+        self.failed_hosts = set()
```

Comment:
Should we also have a `self.tracebacks = set()` here?

Reply:
what would be the purpose? Long term I think it'd be nice to just have a record of everything that was run, with what parameters and the result. Something you can actually dump for debugging purposes.

Reply:
Just to have the tracebacks being saved so that people can collect all of them from the same location. This can of course be handled for each task too.

Reply:
I don't think we should do that for tracebacks explicitly, we already have a mechanism for that in the `AggregatedResult` and I think we need a more generic way of handling this. See #63